File: lib/core/aegis/providers/base_mock_provider.dart
import '../domain/ai_provider.dart';
import '../domain/classification_result.dart';
import '../domain/execution_status.dart';
import '../domain/provider_request.dart';
import '../domain/provider_response.dart';

abstract class BaseMockProvider implements AIProvider {
  const BaseMockProvider();

  @override
  bool get isMock => true;

  Duration get simulatedLatency => const Duration(milliseconds: 450);

  @override
  Future<ProviderResponse> execute(
    ProviderRequest request, {
    required ClassificationResult classification,
  }) async {
    final started = DateTime.now();
    await Future<void>.delayed(simulatedLatency);
    final durationMs = DateTime.now().difference(started).inMilliseconds;

    return ProviderResponse(
      taskType: taskType,
      providerId: id,
      providerName: name,
      executionStatus: ExecutionStatus.completed,
      content: buildContent(request, classification),
      durationMs: durationMs,
    );
  }

  String buildContent(ProviderRequest request, ClassificationResult classification);
}

File: backend/src/core/aegis/providers/base-mock-provider.js
import { ExecutionStatus } from '../task-types.js';

export class BaseMockProvider {
  constructor({ id, name, taskType, latencyMs = 450 }) {
    this.id = id;
    this.name = name;
    this.taskType = taskType;
    this.latencyMs = latencyMs;
    this.isMock = true;
  }

  async execute(request, classification) {
    await new Promise((r) => setTimeout(r, this.latencyMs));
    return {
      taskType: this.taskType,
      providerId: this.id,
      providerName: this.name,
      executionStatus: ExecutionStatus.COMPLETED,
      content: this.buildContent(request, classification),
      isMock: true,
    };
  }

  buildContent(_request, _classification) {
    throw new Error('buildContent must be implemented');
  }
}
